### PR TITLE
populate tm_yday in ds3231_get_time()

### PR DESCRIPTION
**Pull request by hezy0426**

### CODE REVIEW

File: components/ds3231/ds3231.c [hunk 1]
```diff
@@ -87,6 +87,30 @@ static uint8_t dec2bcd(uint8_t val)
     return ((val / 10) << 4) + (val % 10);
 }
 
+#include <stdio.h>
```
Comment (UncleRus) [inline components/ds3231/ds3231.c:90]: move this `#include` to the up. All `includes` must be grouped in one place

File: components/ds3231/ds3231.c [hunk 2]
```diff
@@ -87,6 +87,30 @@ static uint8_t dec2bcd(uint8_t val)
...
+
+// Function to check if a year is a leap year
+int isLeapYear(int year) { return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0); }
```
Comment (UncleRus) [inline components/ds3231/ds3231.c:93]: 1. No CamelCase allowed.
2. No one-line functions format allowed
3. This is static function
4. This is inline function

File: components/ds3231/ds3231.c [hunk 3]
```diff
@@ -87,6 +87,30 @@ static uint8_t dec2bcd(uint8_t val)
...
+
+// Function to convert year, month, and day to days since January 1st
+int daysSinceJanuary1st(int year, int month, int day)
```
Comment (UncleRus) [inline components/ds3231/ds3231.c:96]: 1. No CamelCase allowed
2. It's static
3. It's inline

File: components/ds3231/ds3231.c [hunk 4]
```diff
@@ -87,6 +87,30 @@ static uint8_t dec2bcd(uint8_t val)
...
+{
+    int daysPerMonth[] = { 0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
```
Comment (UncleRus) [inline components/ds3231/ds3231.c:98]: 1. CamelCase
2. This is constant, so declare it `static const ....`

**Review by UncleRus — CHANGES_REQUESTED** [review]

File: components/ds3231/ds3231.c (hunk 1 above)
Comment (hezy0426) [reply components/ds3231/ds3231.c]: Include moved

File: components/ds3231/ds3231.c (hunk 3 above)
Comment (hezy0426) [reply components/ds3231/ds3231.c]: This function is updated to static inline

File: components/ds3231/ds3231.c (hunk 4 above)
Comment (hezy0426) [reply components/ds3231/ds3231.c]: The code is updated to use them as static const arrays.

**Review by UncleRus — APPROVED** [review]

**Comment by UncleRus** [comment]
I haven't read the code much, but I have a vague suspicion about this part:
```C
...
    for (int i = 1; i < month; i++)
    {
        days += *ptr;
        ptr++;
    }
...
```

The fact is that in the `struct tm` month is 0-based:
изображение

If everything is OK and I'm worrying in vain - tell me and I'll merge the PR into the master.

**Comment by hezy0426** [comment]
Hi sir,

Amazing catch there! I have updated the code so that it will work with 0 based tm_mon. 
Quick and random screenshot to show you the result of printing 2023/12/31. 
image

**Comment by UncleRus** [comment]
Thank you!

**Pull request merged**